### PR TITLE
Add UI-related antigravity changes

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Add support for new antigravity table added in https://github.com/pi-hole/pi-hole/pull/5330

This is a follow-up on #2618 which was force-closed and cannot be re-opened.

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Comment by PromoFaux** [comment]
He no worky :(

image

Payload being generated:

```json
{
  "address": [
    "https://raw.githubusercontent.com/hagezi/dns-blocklists/main/adblock/whitelist.txt"
  ],
  "comment": "test",
  "type": "allow"
}
```

**Comment by DL6ER** [comment]
Oh, yeah, sorry. This is an unexpected regression of me pushing the related changes for https://github.com/pi-hole/FTL/pull/1629 into the same branch. I will revert the last commit in here and open a new PR for the latter.

**Comment by yubiuser** [comment]
As the last commit addresses RegexID, I'll report a small visual glitch when a RegEx is trigged by CNAME
Screenshot at 2023-09-14 21-01-34

**Comment by rdwebdesign** [comment]
> I'll report a small visual glitch when a RegEx is trigged by CNAME

This is a design issue. Maybe we can improve the layout of this "details" section (using flex columns or making some other changes) to avoid this overlapping.

**Comment by DL6ER** [comment]
You know, I am by no means a web designer. Concerning web design, I'm all in "Adam mode" (that is: throw several things against a wall and, as soon as something sticks, run away as fast as possible).

File: scripts/pi-hole/js/groups-adlists.js [hunk 1]
```diff
@@ -480,28 +496,29 @@ function delItems(ids) {
     });
 }
 
-function addAdlist() {
+function addAdlist(event) {
```
Comment (yubiuser) [inline scripts/pi-hole/js/groups-adlists.js:500]:
```suggestion
function addList(event) {
```

File: scripts/pi-hole/js/groups-adlists.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ var GETDict = {};
 $(function () {
   GETDict = utils.parseQueryString();
 
-  $("#btnAdd").on("click", addAdlist);
+  $("#btnAddAllow").on("click", { type: "allow" }, addAdlist);
```
Comment (yubiuser) [inline scripts/pi-hole/js/groups-adlists.js:16]:
```suggestion
  $("#btnAddAllow").on("click", { type: "allow" }, addList);
```

File: scripts/pi-hole/js/groups-adlists.js [hunk 3]
```diff
@@ -13,7 +13,8 @@ var GETDict = {};
...
+  $("#btnAddBlock").on("click", { type: "block" }, addAdlist);
```
Comment (yubiuser) [inline scripts/pi-hole/js/groups-adlists.js:17]:
```suggestion
  $("#btnAddBlock").on("click", { type: "block" }, addList);
```

**Review by yubiuser — COMMENTED** [review]
In general: are we using the term "blocklist", "adlist" or "denylist" in v6?

____

L 37 
`placeholder="Adlist description (optional)` > `placeholder="List description (optional)`

L44 
`after modifying your adlists.` > `after modifying your lists`

**Comment by DL6ER** [comment]
> In general: are we using the term "blocklist", "adlist" or "denylist" in v6?

"Blocklist" for gravity domains, "denylist" for manually added domains on the "domainlist" of type "denied" (may be exact or regex). I don't think "adlist" should be used anywhere as the lists *block* and maybe more than just ads (e.g., adult content, etc.).

**Comment by PromoFaux** [comment]
Interesting behaviour when toggling an allowlist from the web interface:

fucky

**Comment by rdwebdesign** [comment]
After the last commits here and FTL, it's working:
fixed

**Review by rdwebdesign — APPROVED** [review]

**Pull request merged**